posts/using-napari-for-cosmx-data/index.qmd

```diff
@@ -0,0 +1,525 @@
+
+knitr::include_graphics("./figures/stitch-widget3.png")
+```
+
+Next, select the location where you want the plugin to return the napari-ready files.
+It is recommend not to store it in the same location as the raw data. Here, I'm 
+pointing it to a location adjacent to the raw data that I named `two_fovs_napari_files`
+(@fig-stitch-widget4).
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-stitch-widget4"
+#| fig-cap: "Select output folder."
+
+knitr::include_graphics("./figures/stitch-widget4.png")
+```
+
+Finally, click `Stitch`. Note: currently there is no refreshing or printing of
+messages. Please do not click `Stitch` more than once. You may see napari become
+unresponse, see the "spinning beachball" (Mac), _etc._ Depending on the number 
+of FOVs, computer configuration, and analyte type, this can take several minutes. 
+Once complete, you should
+see messages that resemble that of @fig-stitch-widget5. If you see the last line
+`See output folder for results`, you successfully converted the raw data into napari-ready files!
+
+To view the results, simply close napari, reopen it, and drag your newly created results
+into the application. 
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-stitch-widget5"
+#| fig-cap: "Messages from a successfully completed stitching run."
+
+knitr::include_graphics("./figures/stitch-widget5.png")
+```
+
+
+
+## Adding and viewing metadata {#sec-adding-metadata} 
+
+While we'll have dedicated posts that discuss tips and tricks for using the
+`napari-cosmx` plugin, here we'll discuss one of the most powerful uses: viewing 
+cells types. 
+
+While not needed for the basic stitching, the Seurat file that is downloaded
+from AtoMx can containing important cell-level information. For example, 
+if cell typing was performed in AtoMx, each cell will have a label with it's cell
+type. 
+
+In this section, I'll show you the basic principle for converting the meta data
+within the Seurat object in to a csv file that can be understood by `napari-cosmx`. 
+Users should have a basic understanding of R in order to use this feature. We'll
+also need to switch our example dataset since the minimal two-FOV example dataset
+was from raw data and not analyzed in AtoMx so we don't have any cell-level cell type
+information. Here, the specific column of interest will have the prefix `RNA_nbclust` and suffix
+`clusters`. In the code below, we'll change that name to simply `cell_types`. We'll
+also need a column named `cell_ID` in the metadata column. We need to write
+the metadata columns to a file specifically named `_metadata.csv` and have that
+file located in the napari-ready folder.  
+
+
+```{r}
+#| label: extracting-seurat-data
+#| echo: true
+#| eval: false
+#| message: true
+#| warning: false
+
+# This is R code
+library(Seurat)
+library(plyr)
+library(dplyr)
+# sem_path will be wherever you downloaded
```

Comment:
... you downloaded _your seurat object_

Reply:
done: 4a912bed94c58c089b5e760f79d420da318d7061